### ISSUE TITLE
Window receives two consecutive `NOTIFICATION_WM_WINDOW_FOCUS_OUT`

### ISSUE THREAD

**Issue by Sauermann** [godotengine/godot#78346]
### Godot version

v4.1.beta.custom_build [9b4d36762]

### System information

Debian Linux 12

### Issue description

Consider the following scene-tree:
- Root Window
  - outer embedded Window
    - inner embedded Window

With a series of mouse-clicks one can send two consecutive `NOTIFICATION_WM_WINDOW_FOCUS_OUT` to the outer embedded window, without any `NOTIFICATION_WM_WINDOW_FOCUS_IN` in-between.

I would expect, that a Window receives in no situation two consecutive `NOTIFICATION_WM_WINDOW_FOCUS_OUT`.

### Steps to reproduce

1. Load and run MRP
2. Click LMB in the outer embedded window, but outside of the inner embedded window
3. Click LMB in the inner embedded window (now the outer window receives a `NOTIFICATION_WM_WINDOW_FOCUS_OUT`)
4. Click LMB outside of the outer embedded window (now the outer window receives a `NOTIFICATION_WM_WINDOW_FOCUS_OUT`)

### Minimal reproduction project

[BugDoubleWindow.zip](https://github.com/godotengine/godot/files/11772914/BugDoubleWindow.zip)

**Comment by Rindbee** [comment]
In these two orders: `root -> outer -> root -> inner` and `root -> inner -> root -> outer`.

https://github.com/godotengine/godot/assets/30386067/bce6d84a-8958-4bfa-97f8-e77eca1b1250




This issue is more like the inner window does not notify `NOTIFICATION_WM_WINDOW_FOCUS_IN` / `NOTIFICATION_WM_WINDOW_FOCUS_OUT`.

**Comment by Sauermann** [comment]
I agree, that this is also a problem. That will however not get resolved by the linked PR.

**Comment by Sauermann** [comment]
Another symptom of this behavior is, that the root window receives a `focus_exited` signal, even though it is still focused in the window manager.
MRP: [BugWindowFocus.zip](https://github.com/godotengine/godot/files/11781215/BugWindowFocus.zip)
1. Load and run MRP
2. Click in embedded Window, Click in background, and repeat
